astropy/io/fits/hdu/hdulist.py

```diff
@@ -1136,6 +1137,14 @@ def _read_next_hdu(self):
                                 fileobj.seek(offset, os.SEEK_SET)
 
                         hdu = _BaseHDU.readfrom(fileobj, **kwargs)
+                    except IOError:
+                        # Close the file: see
+                        # https://github.com/astropy/astropy/issues/6168
+                        #
+                        # TODO: is this a sensible place to close things?
```

Comment:
I would remove these question like todos from the code, and discuss it in the PR itself